tests/functional/testAddons.js

```diff
@@ -305,4 +305,105 @@ describe('Addons', function() {
     await vpn.waitForQueryAndClick(
         queries.screenMessaging.messageItem('message_disabled').visible());
   });
+
+  describe('test message_upgrade_to_annual_plan addon condition', async () => {
+
+    const testCases = [
+      [() => Date.now() - 1000 * 60 * 60 * 24 * 13,
+             "",
+             false, 
+             '13 days after subscription created'],
+      [() => Date.now() - 1000 * 60 * 60 * 24 * 14,
+             "time",
+             true, 
+             '14 days after subscription created'],
+      [() => Date.now() - 1000 * 60 * 60 * 24 * 15,
+             "yesterday",
+             true, 
+             '15 days after subscription created'],
+      [() => Date.now() - 1000 * 60 * 60 * 24 * 86,
+             "date",
+             true, 
+             '86 days after subscription created'],
+      //Fails by an hour (due to daylight savings time?)
+      // [() => Date.now() - 1000 * 60 * 60 * 24 * 87,
+      //        "time",
+      //        true, 
+      //        '87 days after subscription created'],
+      [() => Date.now() - 1000 * 60 * 60 * 24 * 88,
+             "yesterday",
+             true, 
+             '88 days after subscription created'],
+    ];
+
+    const getNextTestCase = testCases[Symbol.iterator]();
+    function setNextSubscriptionStarted(ctx) {
+      const mockDetails = {...SubscriptionDetails};
+      const nextTestCase = getNextTestCase.next().value;
+
+      if (nextTestCase) {
+        const [createdAt] = nextTestCase;
+        // We are faking a Stripe subscription, so this value is expected to be
+        // in seconds.
+        mockDetails.subscription.created = createdAt() / 1000;
+        mockDetails.subscription._subscription_type = "web";
+        mockDetails.plan.interval = "month";
+
+        ctx.guardianSubscriptionDetailsCallback = () => {
+          ctx.guardianOverrideEndpoints.GETs['/api/v1/vpn/subscriptionDetails']
+              .status = 200;
+          ctx.guardianOverrideEndpoints.GETs['/api/v1/vpn/subscriptionDetails']
+              .body = mockDetails;
+        };
+      }
+    }
+
+    // We call this once before all tests to set up the first test,
+    // we can't use beforeEach because that is executed after the guardian
+    // endpoints are overriden.
+    //
+    // We need to setup for the next test before it even starts for the
+    // overrides to apply.
+    setNextSubscriptionStarted(this.ctx);
```

Comment:
Isn't it possible to do `before` or `beforeAll`?

Reply:
the comment above says 

> // We call this once before all tests to set up the first test,
    // we can't use beforeEach because that is executed after the guardian
    // endpoints are overriden.

Reply:
I know but I wonderd if `before` and `beforeAll` would be called at an appropriate time.